mobile_verifier/src/reward_shares.rs

```diff
@@ -642,7 +647,110 @@ impl CoverageShares {
     pub fn test_hotspot_reward_shares(&self, hotspot: &RadioId) -> Decimal {
         self.coverage_points(hotspot)
             .expect("reward shares for hotspot")
-            .reward_shares
+            .total_points()
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct DataTransferAndPocAllocatedRewardBuckets {
+    pub data_transfer: Decimal,
+    pub poc: Decimal,
+    pub boosted_poc: Decimal,
+}
+
+impl DataTransferAndPocAllocatedRewardBuckets {
+    pub fn new(epoch: &Range<DateTime<Utc>>) -> Self {
+        let duration = epoch.end - epoch.start;
+        let total_emission_pool = get_total_scheduled_tokens(duration);
+
+        Self {
+            data_transfer: total_emission_pool * MAX_DATA_TRANSFER_REWARDS_PERCENT,
+            poc: total_emission_pool * POC_REWARDS_PERCENT,
+            boosted_poc: total_emission_pool * BOOSTED_POC_REWARDS_PERCENT,
+        }
+    }
+
+    pub fn new_poc_only(epoch: &Range<DateTime<Utc>>) -> Self {
+        let duration = epoch.end - epoch.start;
+        let total_emission_pool = get_total_scheduled_tokens(duration);
+
+        let poc = total_emission_pool * POC_REWARDS_PERCENT;
+        let data_transfer = total_emission_pool * MAX_DATA_TRANSFER_REWARDS_PERCENT;
+
+        Self {
+            data_transfer: dec!(0),
+            poc: poc + data_transfer,
+            boosted_poc: total_emission_pool * BOOSTED_POC_REWARDS_PERCENT,
+        }
+    }
+
+    pub fn total_poc(&self) -> Decimal {
+        self.poc + self.boosted_poc
+    }
+
+    /// Rewards left over from Data Transfer rewards go into the POC pool. They
+    /// do not get considered for boosted POC rewards if the boost shares
+    /// surpasses the 10% allocation.
+    pub fn handle_unallocated_data_transfer(&mut self, unallocated_data_transfer: Decimal) {
+        self.poc += unallocated_data_transfer;
+    }
+}
+
+/// Given the rewards allocated for Proof of Coverage and all the radios
+/// participating in an epoch, we can calculate the rewards per share for a
+/// radio.
+///
+/// Reference:
+/// HIP 122: Amend Service Provider Hex Boosting
+/// https://github.com/helium/HIP/blob/main/0122-amend-service-provider-hex-boosting.md
+#[derive(Debug)]
+struct CalculatedPocRewardShares {
+    normal: Decimal,
+    boost: Decimal,
+}
+
+impl CalculatedPocRewardShares {
+    fn new<'a>(
+        allocated_rewards: DataTransferAndPocAllocatedRewardBuckets,
+        radios: impl Iterator<Item = &'a coverage_point_calculator::CoveragePoints>,
+    ) -> Option<Self> {
```

Comment:
you can elide this lifetime: 
```suggestion
    fn new(
        allocated_rewards: DataTransferAndPocAllocatedRewardBuckets,
        radios: impl Iterator<Item = &'_ coverage_point_calculator::CoveragePoints>,
    ) -> Option<Self> {
```

Reply:
This rabbit hole reaches depths I do not understand.
___
```rs
    fn new(
        allocated_rewards: DataTransferAndPocAllocatedRewardBuckets,
        radios: impl Iterator<Item = &'_ coverage_point_calculator::CoveragePoints>,
    ) -> Option<Self>
```
``error[E0658]: anonymous lifetimes in `impl Trait` are unstable``
___
```rs
fn new(
        allocated_rewards: DataTransferAndPocAllocatedRewardBuckets,
        radios: impl for<'a> Iterator<Item = &'a coverage_point_calculator::CoveragePoints>,
    ) -> Option<Self>
```
``error[E0582]: binding for associated type `Item` references lifetime `'a`, which does not appear in the trait input types``
___
```rs
    fn new<R>(
        allocated_rewards: DataTransferAndPocAllocatedRewardBuckets,
        radios: R,
    ) -> Option<Self>
    where
        for<'a> R: Iterator<Item = &'a coverage_point_calculator::CoveragePoints>,
```
``error[E0582]: binding for associated type `Item` references lifetime `'a`, which does not appear in the trait input types``

Reply:
ah, just do the original thing then